subiquitycore/async_helpers.py

```diff
@@ -117,3 +118,22 @@ def done(self):
         if self.task is None:
             return False
         return self.task.done()
+
+
+def exclusive(coroutine_function):
+    """Can be used to decorate a coroutine function that we do not want to run
+    multiple times concurrently. It uses a lock internally.
+    If the caller needs to know when the decorated coroutine starts executing
+    (i.e., when it has acquired the exclusive lock), they can pass an
+    asyncio.Event as the "started_event" keyword-only argument.
+    """
+    lock = asyncio.Lock()
+
+    async def wrapped(*args, started_event: Optional[asyncio.Event] = None, **kwargs):
```

Comment:
Does it make sense to have started_event be kwarg-only?

Reply:
I think so? I mean, if I change it to not kwarg-only, it would have to be the first positional argument, right? I'm slightly confused.

Reply:
Oh ignore me I misread that it already is :-)